### ISSUE TITLE
troposphere-2.7.1 fails to install with setuptools-58

### ISSUE THREAD

**Issue by zt-initech** [cloudtools/troposphere#1952]
If you first delete the cached wheel file:

```
find ~/.cache/pip/wheels -name troposphere-2.7.1-py3-none-any.whl -exec rm -rf {} \;
```

and then create a new virtualenv with python3.8, update pip, update setuptools and install wheel:

```
python3.8 -mvenv venv
source venv/bin/activate
pip install --upgrade pip
pip install --upgrade setuptools
pip install --upgrade wheel
pip install --upgrade troposphere\<3
```

it fails like so:

```
Collecting troposphere<3
  Using cached troposphere-2.7.1.tar.gz (205 kB)
    ERROR: Command errored out with exit status 1:
     command: /home/zeev/threat-landscape/venv_tropo/bin/python3.8 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-bbon_gw1/troposphere_7caad3cd9aa841e9a57b5f18a2fe1f88/setup.py'"'"'; __file__='"'"'/tmp/pip-install-bbon_gw1/troposphere_7caad3cd9aa841e9a57b5f18a2fe1f88/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-v6jdg1qx
         cwd: /tmp/pip-install-bbon_gw1/troposphere_7caad3cd9aa841e9a57b5f18a2fe1f88/
    Complete output (1 lines):
    error in troposphere setup command: use_2to3 is invalid.
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/a7/f5/b066cea6022142792207d356d9183b3812d01e27910023438f354aa98b43/troposphere-2.7.1.tar.gz#sha256=ea2e5f2f82c224eaa1414a008b1939aae124c3e3e1dd993301968f155b333bd7 (from https://pypi.org/simple/troposphere/) (requires-python:>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Downloading troposphere-2.7.0.tar.gz (207 kB)
```

and tries lower versions until it reaches troposphere-0.1.2.

On the other hand, if you install setuptools-57.X, it works fine.

(We pin troposphere<3 because of zappa)

**Comment by BassonJulien** [comment]
+1

**Comment by markpeek** [comment]
This is due to a [breaking change](https://setuptools.readthedocs.io/en/latest/history.html#v58-0-0) in setuptools 58.0.0 which removes 2to3. You could pin setuptools to be an earlier version.

A zappa compatibility issue came up after I release troposphere 3.0.0 in early July. I commented it was documented in the troposphere release notes, had been marked deprecated for quite some time, and is an [easy fix](https://github.com/zappa/Zappa/issues/998#issuecomment-874776331). It also came up in a PR review where I [commented](https://github.com/zappa/Zappa/pull/999#discussion_r685497110) it could safely be changed for both Python 2 and Python 3 as it is backwards compatible. Really the zappa maintainers need to make the simple change and release a new version.

**Comment by markpeek** [comment]
@zt-initech @BassonJulien 
I [commented](https://github.com/zappa/Zappa/pull/1029#issuecomment-916346784) on this PR. Feel free to +1 it.

**Comment by markpeek** [comment]
zappa merged the change to upgrade to troposphere 3.0+. Closing.
